lib/auth/auth.go

```diff
@@ -673,11 +673,32 @@ func (a *AuthServer) ValidateOIDCAuthCallback(q url.Values) (*OIDCAuthResponse,
 		return response, nil
 	}
 
-	user, err := a.Identity.GetUserByOIDCIdentity(services.OIDCIdentity{
+	var user services.User
+
+	user, err = a.Identity.GetUserByOIDCIdentity(services.OIDCIdentity{
 		ConnectorID: req.ConnectorID, Email: ident.Email})
 	if err != nil {
-		return nil, trace.Wrap(err)
+		stoken, err := a.CreateSignupToken(&services.TeleportUser{
```

Comment:
I'm wondering if it would be worth it to just add something to create a user that doesn't require a token?  Then we wouldn't need to special case the token skipping logic in new_web_user.CreateUserWithToken.  We're really asking to just be able to create a user so why not add a method for this.